### ISSUE TITLE
questions

### ISSUE THREAD

**Issue by vgoklani** [shama/view-list#5]
Nice library! I have a few quick questions:
1. Is there a way to prepend new items to the top, and have the elements fall off the bottom when the list size > some cutoff? like a twitter stream? 
2. Do you plan to support mobile rendering. The demo version renders on my iPhone, but it's not so smooth.
3. How would you implement filtering? Do you filter the array first, and then re-render the table?

Thanks!

**Comment by shama** [comment]
1. Modify the array and render:
   
   ```
   data.unshift('item on top')
   data = data.slice(0, 15)
   viewlist.render(data)
   ```
2. It just renders DOM elements. It may not be optimized for mobile but it is supported. Rendering less or differently should produce smoother results but depends on the app.
3. Yep, modify the array and render:
   
   ```
   data = data.filter(function (row) {
     return row === 'row we want'
   })
   viewlist.render(data)
   ```

**Comment by vgoklani** [comment]
Thanks, I'm able to prepend the data!

One more quick question, I would like the prepended data to be rendered automatically *only* when the first element is visible in the DOM. If a user is scrolling through the list, then I don't want the table to shift, (it should be static so it's easy to read). It should only render new elements when the top-most row is visible.

I could hack a solution by looking at the "scroll" field from the EventListener and check what's being displayed and compare it to the array in memory, but I was wondering if you had a more elegant solution.

Thanks!

**Comment by vgoklani** [comment]
Thanks!

Is it a simple process to extend this approach to a full table (with columns)? Also, how would you optimize for mobile?

**Comment by shama** [comment]
Yep! By default it uses `ul/li` but you can change that by setting the `tagName` to `'table'` and  `childTagName` to `'tr'`. Then return rows with columns on `eachrow`.

I recommend extending this element if you're planning on something more sophisticated like that. Here is an example of someone else extending this element: https://github.com/sethvincent/data-cards

**Comment by vgoklani** [comment]
Thanks, worked perfectly! Unfortunately the "scroll" EventListener is no longer working. It stops emitting when I change the ul/li to table/tr in index.js. Any suggestions?

Thanks!

**Comment by vgoklani** [comment]
Almost have it, what do you mean by this: "Then return rows with columns on eachrow."
